docs/source/guides/inference-multimodal.mdx

```diff
@@ -0,0 +1,175 @@
+with open(image_path, "rb") as f:
+    base64_image = base64.b64encode(f.read()).decode("utf-8")
+image_url = f"data:image/jpeg;base64,{base64_image}"
+
+chat_completion = client.chat.completions.create(
+    messages=[
+        {
+            "role": "user",
+            "content": [
+                {
+                    "type": "text",
+                    "text": "Describe this image in detail.",
+                },
+                {
+                    "type": "image_url",
+                    "image_url": {"url": image_url},
+                },
+            ],
+        },
+    ],
+    temperature=0.7,
+    max_tokens=128,
+)
+print(chat_completion.choices[0].message.content)
+```
+
+#### `openai`
+
+Install the OpenAI package:
+```bash
+pip install --upgrade openai
+```
+
+Then use it similarly to the HuggingFace client:
+
+```python
+from openai import OpenAI
+import base64
+
+client = OpenAI(base_url="http://localhost:8080/v1/", api_key="-")
+
+# Using a URL or base64-encoded image
+image_url = "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"  # or your base64 data URL
+chat_completion = client.chat.completions.create(
+    model="your-model",
+    messages=[
+        {
+            "role": "user",
+            "content": [
+                {
+                    "type": "text",
+                    "text": "Describe this image in detail.",
+                },
+                {
+                    "type": "image_url",
+                    "image_url": {"url": image_url},
+                },
+            ],
+        },
+    ],
+    temperature=0.7,
+    max_tokens=128,
+)
+print(chat_completion.choices[0].message.content)
+```
+
+### cURL
+
+You can also make multimodal requests using cURL. Here's an example using an image URL:
+
+```bash
+curl http://localhost:8080/v1/chat/completions \
+    -X POST \
+    -d '{
+        "model":"your-model",
+        "messages":[{
+            "role":"user",
+            "content":[
+                {
+                    "type":"text",
+                    "text":"Describe this image."
+                },
+                {
+                    "type":"image_url",
+                    "image_url":{"url":"https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"}
+                }
+            ]
+        }],
+        "temperature":0.7,
+        "max_tokens":128
+    }' \
+    -H 'Content-Type: application/json'
+```
+
+## Supported Image Formats
+
+The following image formats are supported:
+- JPEG/JPG
+- PNG
+- GIF (first frame only)
+- WebP
+
+<Tip>
+When using base64-encoded images, make sure to include the correct MIME type in the data URL (e.g., `data:image/jpeg;base64,` for JPEG images).
+</Tip>
+
+## Best Practices
+
+1. **Image Size**: While there's no strict limit on image dimensions, it's recommended to resize large images before sending them to reduce bandwidth usage and processing time.
```

Comment:
is there a recommended range of image dimensions that makes sense for most multimodal models?